INFO 2025-03-19 14:52:21,194 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:52:48,861 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\config\settings.py changed, reloading.
INFO 2025-03-19 14:52:51,734 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:53:29,695 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:54:37,976 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\payments\views.py changed, reloading.
INFO 2025-03-19 14:54:40,213 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:55:08,579 views Processing bKash payment for order 5
INFO 2025-03-19 14:55:08,585 views Callback URL: http://127.0.0.1:8000/payments/bkash/callback/
INFO 2025-03-19 14:55:08,587 views Session data: {'cart_id': '38ba548f-3e39-432d-8d01-351955de1b0d', '_auth_user_id': '3', '_auth_user_backend': 'django.contrib.auth.backends.ModelBackend', '_auth_user_hash': '837f3cf7444deffc01048f547f58dd056402c83f8b78c312d830ceebb33ece09', 'order_id': 5, 'checkout_data': {'full_name': 'Shahariar Hasan', 'email': 'infodomhost@gmail.com', 'phone': '01710516652', 'address_line_1': '85/2, Borobag', 'address_line_2': '', 'city': 'Mirpur-2', 'postal_code': '1216', 'payment_method': 'bkash'}}
INFO 2025-03-19 14:55:08,588 services BkashPaymentService initialized with: app_key=6rAS..., username=01612611770, base_url=https://tokenized.pay.bka.sh/v1.2.0-beta
INFO 2025-03-19 14:55:08,590 services Getting bKash token with payload: {'app_key': '6rASpj5aUtgin9UJRDl3l7vrtc', 'app_secret': 'gxiav9mLDLdxmJ2vR9tzIdpit7lGYzkAdcuE0MjSh8IdPoYQMmuj'}
DEBUG 2025-03-19 14:55:08,598 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 14:55:09,282 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/token/grant HTTP/1.1" 200 2759
INFO 2025-03-19 14:55:09,285 services Token response status: 200
INFO 2025-03-19 14:55:09,287 services Token received successfully
INFO 2025-03-19 14:55:09,289 services Creating bKash payment with: amount=1200.00, invoice=ORD2503191831D7
DEBUG 2025-03-19 14:55:09,295 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 14:55:09,590 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/create HTTP/1.1" 200 78
INFO 2025-03-19 14:55:09,593 services Create payment response status: 200
INFO 2025-03-19 14:55:09,599 services Payment creation successful with paymentID: None
INFO 2025-03-19 14:55:09,602 views bKash response: {'statusCode': '2048', 'statusMessage': 'Invalid Payer Reference'}
ERROR 2025-03-19 14:55:09,605 views Failed to create payment. Response: {'statusCode': '2048', 'statusMessage': 'Invalid Payer Reference'}
INFO 2025-03-19 14:58:12,660 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\config\settings.py changed, reloading.
INFO 2025-03-19 14:58:14,416 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:58:46,730 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\payments\services.py changed, reloading.
INFO 2025-03-19 14:58:48,414 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:59:01,834 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\payments\views.py changed, reloading.
INFO 2025-03-19 14:59:03,371 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 14:59:28,036 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\payments\urls.py changed, reloading.
INFO 2025-03-19 14:59:30,483 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:01:25,717 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\orders\views.py changed, reloading.
INFO 2025-03-19 15:01:27,256 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:04:13,322 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\orders\views.py changed, reloading.
INFO 2025-03-19 15:04:15,030 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:05:31,592 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\orders\views.py changed, reloading.
INFO 2025-03-19 15:05:32,299 autoreload Watching for file changes with StatReloader
DEBUG 2025-03-19 15:06:00,522 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 15:06:00,995 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/token/grant HTTP/1.1" 200 2759
DEBUG 2025-03-19 15:06:01,000 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 15:06:01,200 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/create HTTP/1.1" 200 996
WARNING 2025-03-19 15:16:16,984 log Bad Request: /cart/add/4/
DEBUG 2025-03-19 15:20:41,136 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 15:20:41,611 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/token/grant HTTP/1.1" 200 2759
DEBUG 2025-03-19 15:20:41,613 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-19 15:20:41,802 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/create HTTP/1.1" 200 996
INFO 2025-03-19 15:25:52,200 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:58:06,028 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\orders\admin.py changed, reloading.
INFO 2025-03-19 15:58:08,573 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:58:43,004 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\products\models.py changed, reloading.
INFO 2025-03-19 15:58:44,609 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 15:59:19,617 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\products\admin.py changed, reloading.
INFO 2025-03-19 15:59:21,273 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 16:02:31,504 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\orders\admin.py changed, reloading.
INFO 2025-03-19 16:02:34,233 autoreload Watching for file changes with StatReloader
INFO 2025-03-19 16:10:05,108 autoreload C:\Users\Shafayet\OneDrive\Desktop\project_root\config\urls.py changed, reloading.
INFO 2025-03-19 16:10:06,989 autoreload Watching for file changes with StatReloader
ERROR 2025-03-19 16:12:41,165 log Internal Server Error: /admin/products/productvariant/
Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\options.py", line 718, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\views\decorators\cache.py", line 80, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\sites.py", line 241, in inner
    return view(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\options.py", line 1999, in changelist_view
    cl = self.get_changelist_instance(request)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\options.py", line 866, in get_changelist_instance
    return ChangeList(
        request,
    ...<12 lines>...
        self.search_help_text,
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\views\main.py", line 145, in __init__
    self.queryset = self.get_queryset(request)
                    ~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\views\main.py", line 545, in get_queryset
    ) = self.get_filters(request)
        ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\views\main.py", line 216, in get_filters
    spec = field_list_filter_class(
        field,
    ...<4 lines>...
        field_path=field_path,
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\filters.py", line 217, in create
    return list_filter_class(
        field, request, params, model, model_admin, field_path=field_path
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\filters.py", line 232, in __init__
    self.lookup_choices = self.field_choices(field, request, model_admin)
                          ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\filters.py", line 273, in field_choices
    return field.get_choices(include_blank=False, ordering=ordering)
           ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\fields\__init__.py", line 1077, in get_choices
    (choice_func(x), str(x)) for x in qs
                                      ^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 400, in __iter__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
ERROR 2025-03-19 16:12:48,747 log Internal Server Error: /admin/products/category/
Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_category.parent_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\options.py", line 718, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\views\decorators\cache.py", line 80, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\sites.py", line 241, in inner
    return view(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 48, in _wrapper
    return bound_method(*args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 188, in _view_wrapper
    result = _process_exception(request, e)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\utils\decorators.py", line 186, in _view_wrapper
    response = view_func(request, *args, **kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\options.py", line 2138, in changelist_view
    "selection_note": _("0 of %(cnt)s selected") % {"cnt": len(cl.result_list)},
                                                           ~~~^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_category.parent_id
ERROR 2025-03-19 16:15:27,090 log Internal Server Error: /admin/cart/cart/
Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\options.py", line 676, in get_field
    return self.fields_map[field_name]
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: 'total_price'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\utils.py", line 290, in lookup_field
    f = _get_non_gfk_field(opts, name)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\utils.py", line 330, in _get_non_gfk_field
    field = opts.get_field(name)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\options.py", line 678, in get_field
    raise FieldDoesNotExist(
        "%s has no field named '%s'" % (self.object_name, field_name)
    )
django.core.exceptions.FieldDoesNotExist: Cart has no field named 'total_price'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\fields\related_descriptors.py", line 239, in __get__
    rel_obj = self.field.get_cached_value(instance)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\fields\mixins.py", line 37, in get_cached_value
    return instance._state.fields_cache[self.cache_name]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'product_variant'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_productvariant.price_adjustment

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\templatetags\base.py", line 45, in render
    return super().render(context)
           ~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\library.py", line 258, in render
    _dict = self.func(*resolved_args, **resolved_kwargs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 346, in result_list
    "results": list(results(cl)),
               ~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 322, in results
    yield ResultList(None, items_for_result(cl, res, None))
          ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 313, in __init__
    super().__init__(*items)
    ~~~~~~~~~~~~~~~~^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\templatetags\admin_list.py", line 218, in items_for_result
    f, attr, value = lookup_field(field_name, result, cl.model_admin)
                     ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\utils.py", line 302, in lookup_field
    attr = getattr(obj, name, sentinel)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\cart\models.py", line 17, in total_price
    return sum(item.total_price for item in self.items.all())
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\cart\models.py", line 17, in <genexpr>
    return sum(item.total_price for item in self.items.all())
               ^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\cart\models.py", line 37, in total_price
    if hasattr(self.product_variant, 'price'):
               ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\fields\related_descriptors.py", line 257, in __get__
    rel_obj = self.get_object(instance)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\fields\related_descriptors.py", line 220, in get_object
    return qs.get(self.field.get_reverse_related_filter(instance))
           ~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 645, in get
    num = len(clone)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
    ~~~~~~~~~~~~~~~^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_productvariant.price_adjustment
ERROR 2025-03-19 16:16:06,134 log Internal Server Error: /admin/products/productvariant/add/
Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 220, in _get_response
    response = response.render()
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\response.py", line 92, in rendered_content
    return template.render(context, self._request)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 210, in render
    return template.render(context)
           ~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 173, in render
    return self._render(context)
           ~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 243, in render
    nodelist.append(node.render_annotated(context))
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 327, in render
    return nodelist.render(context)
           ~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 327, in render
    return nodelist.render(context)
           ~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ~~~~~~~~~~~^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1073, in render
    return render_value_in_context(output, context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1050, in render_value_in_context
    value = str(value)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\utils.py", line 79, in __str__
    return self.as_widget()
           ~~~~~~~~~~~~~~^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\boundfield.py", line 108, in as_widget
    return widget.render(
           ~~~~~~~~~~~~~^
        name=self.html_initial_name if only_initial else self.html_name,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
        renderer=self.form.renderer,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\widgets.py", line 278, in render
    context = self.get_context(name, value, attrs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\contrib\admin\widgets.py", line 329, in get_context
    "rendered_widget": self.widget.render(name, value, attrs),
                       ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\widgets.py", line 278, in render
    context = self.get_context(name, value, attrs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\widgets.py", line 764, in get_context
    context = super().get_context(name, value, attrs)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\widgets.py", line 715, in get_context
    context["widget"]["optgroups"] = self.optgroups(
                                     ~~~~~~~~~~~~~~^
        name, context["widget"]["value"], attrs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\widgets.py", line 655, in optgroups
    for index, (option_value, option_label) in enumerate(self.choices):
                                               ~~~~~~~~~^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\forms\models.py", line 1424, in __iter__
    for obj in queryset:
               ^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 518, in _iterator
    yield from iterable
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
        chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size
    )
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        sql, params, many=False, executor=self._execute
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "C:\Users\Shafayet\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
INFO 2025-03-22 02:23:45,552 autoreload Watching for file changes with StatReloader
INFO 2025-03-22 02:25:54,923 autoreload Watching for file changes with StatReloader
ERROR 2025-03-22 02:26:20,458 log Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\core\views.py", line 14, in home
    return render(request, 'home.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
                 ^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
WARNING 2025-03-22 02:26:21,581 log Not Found: /favicon.ico
INFO 2025-03-22 02:37:54,043 autoreload Watching for file changes with StatReloader
ERROR 2025-03-22 02:38:19,505 log Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\core\views.py", line 14, in home
    return render(request, 'home.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
                 ^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
INFO 2025-03-22 02:40:25,830 autoreload C:\Users\mshaf\OneDrive\Desktop\project_root\products\admin.py changed, reloading.
INFO 2025-03-22 02:40:28,447 autoreload Watching for file changes with StatReloader
ERROR 2025-03-22 02:40:37,412 log Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\core\views.py", line 14, in home
    return render(request, 'home.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
                 ^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
INFO 2025-03-22 02:52:15,245 autoreload Watching for file changes with StatReloader
ERROR 2025-03-22 02:52:20,457 log Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such column: products_product.is_active

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\core\views.py", line 14, in home
    return render(request, 'home.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\template\defaulttags.py", line 199, in render
    len_values = len(values)
                 ^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 382, in __len__
    self._fetch_all()
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 1928, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\query.py", line 91, in __iter__
    results = compiler.execute_sql(
              ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\models\sql\compiler.py", line 1574, in execute_sql
    cursor.execute(sql, params)
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project_root\venv\Lib\site-packages\django\db\backends\sqlite3\base.py", line 354, in execute
    return super().execute(query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such column: products_product.is_active
INFO 2025-03-22 02:57:06,047 autoreload Watching for file changes with StatReloader
INFO 2025-03-22 02:57:35,742 autoreload Watching for file changes with StatReloader
DEBUG 2025-03-22 03:00:31,456 Image Importing BlpImagePlugin
DEBUG 2025-03-22 03:00:31,461 Image Importing BmpImagePlugin
DEBUG 2025-03-22 03:00:31,462 Image Importing BufrStubImagePlugin
DEBUG 2025-03-22 03:00:31,464 Image Importing CurImagePlugin
DEBUG 2025-03-22 03:00:31,466 Image Importing DcxImagePlugin
DEBUG 2025-03-22 03:00:31,472 Image Importing DdsImagePlugin
DEBUG 2025-03-22 03:00:31,488 Image Importing EpsImagePlugin
DEBUG 2025-03-22 03:00:31,499 Image Importing FitsImagePlugin
DEBUG 2025-03-22 03:00:31,502 Image Importing FliImagePlugin
DEBUG 2025-03-22 03:00:31,505 Image Importing FpxImagePlugin
DEBUG 2025-03-22 03:00:31,511 Image Image: failed to import FpxImagePlugin: No module named 'olefile'
DEBUG 2025-03-22 03:00:31,514 Image Importing FtexImagePlugin
DEBUG 2025-03-22 03:00:31,517 Image Importing GbrImagePlugin
DEBUG 2025-03-22 03:00:31,518 Image Importing GifImagePlugin
DEBUG 2025-03-22 03:00:31,520 Image Importing GribStubImagePlugin
DEBUG 2025-03-22 03:00:31,522 Image Importing Hdf5StubImagePlugin
DEBUG 2025-03-22 03:00:31,530 Image Importing IcnsImagePlugin
DEBUG 2025-03-22 03:00:31,536 Image Importing IcoImagePlugin
DEBUG 2025-03-22 03:00:31,540 Image Importing ImImagePlugin
DEBUG 2025-03-22 03:00:31,546 Image Importing ImtImagePlugin
DEBUG 2025-03-22 03:00:31,549 Image Importing IptcImagePlugin
DEBUG 2025-03-22 03:00:31,551 Image Importing JpegImagePlugin
DEBUG 2025-03-22 03:00:31,551 Image Importing Jpeg2KImagePlugin
DEBUG 2025-03-22 03:00:31,553 Image Importing McIdasImagePlugin
DEBUG 2025-03-22 03:00:31,555 Image Importing MicImagePlugin
DEBUG 2025-03-22 03:00:31,565 Image Image: failed to import MicImagePlugin: No module named 'olefile'
DEBUG 2025-03-22 03:00:31,566 Image Importing MpegImagePlugin
DEBUG 2025-03-22 03:00:31,567 Image Importing MpoImagePlugin
DEBUG 2025-03-22 03:00:31,574 Image Importing MspImagePlugin
DEBUG 2025-03-22 03:00:31,579 Image Importing PalmImagePlugin
DEBUG 2025-03-22 03:00:31,584 Image Importing PcdImagePlugin
DEBUG 2025-03-22 03:00:31,586 Image Importing PcxImagePlugin
DEBUG 2025-03-22 03:00:31,588 Image Importing PdfImagePlugin
DEBUG 2025-03-22 03:00:31,610 Image Importing PixarImagePlugin
DEBUG 2025-03-22 03:00:31,614 Image Importing PngImagePlugin
DEBUG 2025-03-22 03:00:31,615 Image Importing PpmImagePlugin
DEBUG 2025-03-22 03:00:31,615 Image Importing PsdImagePlugin
DEBUG 2025-03-22 03:00:31,617 Image Importing QoiImagePlugin
DEBUG 2025-03-22 03:00:31,620 Image Importing SgiImagePlugin
DEBUG 2025-03-22 03:00:31,622 Image Importing SpiderImagePlugin
DEBUG 2025-03-22 03:00:31,624 Image Importing SunImagePlugin
DEBUG 2025-03-22 03:00:31,631 Image Importing TgaImagePlugin
DEBUG 2025-03-22 03:00:31,633 Image Importing TiffImagePlugin
DEBUG 2025-03-22 03:00:31,634 Image Importing WebPImagePlugin
DEBUG 2025-03-22 03:00:31,640 Image Importing WmfImagePlugin
DEBUG 2025-03-22 03:00:31,644 Image Importing XbmImagePlugin
DEBUG 2025-03-22 03:00:31,652 Image Importing XpmImagePlugin
DEBUG 2025-03-22 03:00:31,656 Image Importing XVThumbImagePlugin
DEBUG 2025-03-22 03:01:16,351 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-22 03:01:16,866 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/token/grant HTTP/1.1" 200 2759
DEBUG 2025-03-22 03:01:16,870 connectionpool Starting new HTTPS connection (1): tokenized.pay.bka.sh:443
DEBUG 2025-03-22 03:01:17,078 connectionpool https://tokenized.pay.bka.sh:443 "POST /v1.2.0-beta/tokenized/checkout/create HTTP/1.1" 200 996
INFO 2025-03-23 15:50:39,711 autoreload Watching for file changes with StatReloader
WARNING 2025-03-23 15:50:52,028 log Not Found: /favicon.ico
INFO 2025-03-23 15:52:44,124 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 15:52:46,791 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 15:56:38,836 autoreload Watching for file changes with StatReloader
ERROR 2025-03-23 16:01:14,842 log Internal Server Error: /crm/
Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\base.py", line 1008, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 134, in render
    compiled_parent = self.get_parent(context)
                      ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 131, in get_parent
    return self.find_template(parent, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\loader_tags.py", line 109, in find_template
    template, origin = context.template.engine.find_template(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\engine.py", line 163, in find_template
    raise TemplateDoesNotExist(name, tried=tried)
django.template.exceptions.TemplateDoesNotExist: crm/base.html

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\decorators.py", line 17, in wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py", line 113, in dashboard
    return render(request, 'crm/dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 109, in render
    reraise(exc, self.backend)
  File "C:\Users\mshaf\OneDrive\Desktop\project\project_root\venv\Lib\site-packages\django\template\backends\django.py", line 130, in reraise
    raise new from exc
django.template.exceptions.TemplateDoesNotExist: crm/base.html
INFO 2025-03-23 16:04:21,261 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:04:25,687 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:10:17,646 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:10:19,788 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:10:54,151 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\decorators.py changed, reloading.
INFO 2025-03-23 16:10:58,119 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:13:24,303 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:13:26,795 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:14:23,215 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\urls.py changed, reloading.
INFO 2025-03-23 16:14:25,097 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:18:13,124 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:32:19,382 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:32:20,886 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:33:12,336 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:33:14,873 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:33:55,874 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\views.py changed, reloading.
INFO 2025-03-23 16:33:58,561 autoreload Watching for file changes with StatReloader
INFO 2025-03-23 16:51:12,597 autoreload C:\Users\mshaf\OneDrive\Desktop\project\project_root\crm\models.py changed, reloading.
INFO 2025-03-23 16:51:16,491 autoreload Watching for file changes with StatReloader
